modules/reindex/src/test/java/org/elasticsearch/index/reindex/RethrottleTests.java

```diff
@@ -174,6 +171,39 @@ private void testCase(AbstractBulkByScrollRequestBuilder<?, ?> request, String a
                 response.getBatches(), greaterThanOrEqualTo(numSlices));
     }
 
+    private ListTasksResponse rethrottleTask(TaskId taskToRethrottle, float newRequestsPerSecond) {
+        // the task isn't ready to be rethrottled until it has figured out how many slices it will use. if we rethrottle when the task is
+        // in this state, the request will fail. so we try a few times
+        long start = System.nanoTime();
+        List<Throwable> failures = new ArrayList<>();
+        do {
```

Comment:
Right, returning something was the other thing that didn't play well. Before I had it set an AtomicReference which worked but looked a little weird